test/e2e/ic_conditions_metric_test.go

```diff
@@ -0,0 +1,136 @@
+		t.Fatalf("failed to find the ingresscontroller operator pods: %v", err)
+	}
+	operatorPodName := podsList.Items[0].Name
+	if operatorPodName == "" {
+		t.Fatalf("unable to fetch Ingress Operator pod name")
+	}
+	// Delete the operator pod
+	if err := kubeClient.CoreV1().Pods("openshift-ingress-operator").Delete(context.TODO(), operatorPodName, metav1.DeleteOptions{}); err != nil {
+		t.Fatalf("failed to find the ingresscontroller operator pods: %v", err)
+	}
+
+	interval, timeout := 5*time.Second, 5*time.Minute
+	// Wait for new pod to be ready
+	t.Logf("Polling for up to %v to verify that the oprator restart is terminated...", timeout)
+	if err := wait.PollUntilContextTimeout(context.Background(), interval, timeout, false, func(context context.Context) (bool, error) {
+		if podsList, err = kubeClient.CoreV1().Pods("openshift-ingress-operator").List(context, metav1.ListOptions{}); err != nil || len(podsList.Items) == 0 {
+			return false, err
+		}
+		return true, nil
+	}); err != nil {
+		t.Fatalf("Operator pod did not restart in time")
+	}
+	return time.Now()
+}
+
+// TestIngressControllerConditionsMetricAfterRestart verifies that metric ingress_controller_conditions(router,status) is
+// available after an operator pod restart too.
+//
+// This test:
+//
+// 1. Verifies that the metric is available in a normal situation when the operator pod is up&running (i.e. before restart)
+//
+// 2. Restarts the operator pod, waiting it will be available again
+//
+// 3. Repeats the step 1 expecting the same result again and so the presence of the metric
+//
+// NB:
+//  1. this test requires an OpenShift version with the monitoring stack up&running
+//  2. due to the fact that this test is restarting the operator pod it cannot be executed in parallel with other tests
+func TestIngressControllerConditionsMetricAfterRestart(t *testing.T) {
+	metricName := "ingress_controller_conditions"
+
+	// Create a new prometheus client for fetching metrics and dependencies needed
+	kubeConfig, err := config.GetConfig()
+	if err != nil {
+		t.Fatalf("failed to get kube config: %s", err)
+	}
+	kubeClient, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		t.Fatal(err)
+	}
+	routeClient, err := routev1client.NewForConfig(kubeConfig)
+	if err != nil {
+		t.Fatal(err)
+	}
+	prometheusClient, err := metrics.NewPrometheusClient(context.TODO(), kubeClient, routeClient)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Check metric before restart
+	t.Logf("Verifying that in Prometheus metrics there are %s metrics before resart", metricName)
+	// Wait for metrics to be added and set to 0.
+	if err := waitForIngressControllerConditionsMetrics(t, prometheusClient, time.Now().Add(-5*time.Minute)); err != nil {
+		t.Fatalf("failed to fetch expected metrics: %v", err)
+	}
+
+	// Restart operator pod
+	restartTime := restartOperatorPod(t, kubeClient)
```

Comment:
`restartOperator` can use [the global kube client](https://github.com/openshift/cluster-ingress-operator/blob/0cac97af560a68a5f5c0390a529726eb43556d51/test/e2e/operator_test.go#L120).